packages/machine-extractor/src/sky/skyConfigExtractResult.ts

```diff
@@ -17,6 +22,13 @@ export function skyConfigExtractResult({
       const fileText = fileContent.substring(node.start!, node.end!);
       return hashedId(fileText);
     },
+    getNodeSource: (node: t.Node): string => {
+      return fileContent.substring(node.start!, node.end!);
+    },
+    getEnvVariable: (name: string): string | undefined => {
```

Comment:
one last thing here - it likely should be possible to read the value from actual `process.env` too. This way you could override whatever can be found in `.env` or you could just use env vars if you are not using env files at all.

So I'd propose smth like this here, at the top of the function:
```ts
if (process.env[name]) {
  return process.env[name];
}
```